packages/block-editor/src/components/font-appearance-control/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * WordPress dependencies
+ */
+import { CustomSelectControl } from '@wordpress/components';
+import { useMemo } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Control to display unified font style and weight options.
+ *
+ * @param  {Object}   props          Component props.
+ * @param  {Object}   props.value    Currently selected combination of font style and weight.
+ * @param  {Object}   props.options  Object containing weight and style options.
+ * @param  {Function} props.onChange Handles selection change.
+ * @return {WPElement}               Font appearance control.
+ */
+export default function FontAppearanceControl( { value, options, onChange } ) {
+	const { fontStyle, fontWeight } = value;
+	const { fontStyles = [], fontWeights = [] } = options;
+	const hasStylesOrWeights = fontStyles.length > 0 || fontWeights.length > 0;
+
+	// Map font styles and weights to select options.
+	const selectOptions = useMemo( () => {
+		const defaultCombo = { fontStyle: undefined, fontWeight: undefined };
+		const combinedOptions = [
+			{
+				key: 'default',
+				name: __( 'Default' ),
+				style: defaultCombo,
+				presetStyle: defaultCombo,
+			},
+		];
+
+		fontStyles.forEach( ( { name: styleName, slug: styleSlug } ) => {
+			fontWeights.forEach( ( { name: weightName, slug: weightSlug } ) => {
+				combinedOptions.push( {
+					key: `${ weightSlug }-${ styleSlug }`,
+					name:
+						styleSlug === 'normal'
+							? weightName
+							: `${ weightName } ${ styleName }`,
```

Comment:
I guess it may make sense to use sprintf and our translated function __ for i18n, depending on the locale it may make sense to change the order of weightName and styleName.